src/rust/scheduler/scheduler.rs

```diff
@@ -136,12 +136,20 @@ impl Scheduler {
 
         // Generate a new id. If the id is currently in use, keep generating until we find an unused id.
         let mut task_ids: RefMut<HashMap<u64, usize>> = self.task_ids.borrow_mut();
-        let task_id: u64 = loop {
-            let id: u64 = id_gen.next_u64() as u16 as u64;
-            if !task_ids.contains_key(&id) {
-                task_ids.insert(id, index);
-                break id;
+        // If the address space for task ids is close to half full, it will become increasingly difficult to avoid
+        // collisions, so we cap the number of tasks at 16,000.
+        if task_ids.len() > 16000 {
+            panic!("Too many concurrent tasks");
+        }
+        let task_id: u64 = 'get_id: {
+            for _ in 0..100 {
```

Comment:
Could we have a constant for this?

Also, should this number be higher?

Reply:
done